Mage.Server.Plugins/Mage.Player.Human/src/mage/player/human/HumanPlayer.java

```diff
@@ -529,17 +529,34 @@ public boolean choose(Outcome outcome, Target target, Ability source, Game game,
                 required = false;
             }
 
-            java.util.List<UUID> chosen = target.getTargets();
-            options.put("chosen", (Serializable) chosen);
+            UUID responseId = null;
+            boolean canAutochoose = target.getMinNumberOfTargets() == target.getMaxNumberOfTargets() && // Targets must be picked
+                                    target.getNumberOfTargets() - target.getSize() == possibleTargetIds.size(); // Available targets are equal to the number that must be picked
+            if (canAutochoose) {
+                // Targets can be auto-chosen
+                for (UUID possibleTargetId : possibleTargetIds) {
+                    if (!target.getTargets().contains(possibleTargetId)) {
+                        // Make sure to not autopick one that's already been picked, otherwise we may hit an infinite loop
+                        responseId = possibleTargetId;
+                        break;
+                    }
+                }
+            }
 
-            updateGameStatePriority("choose(5)", game);
-            prepareForResponse(game);
-            if (!isExecutingMacro()) {
-                game.fireSelectTargetEvent(getId(), new MessageToClient(target.getMessage(), getRelatedObjectName(source, game)), targetIds, required, getOptions(target, options));
+            if (responseId == null) {
+                java.util.List<UUID> chosen = target.getTargets();
+                options.put("chosen", (Serializable) chosen);
+
+                updateGameStatePriority("choose(5)", game);
+                prepareForResponse(game);
```

Comment:
Should these be called even if the choice is auto-made?

Reply:
Nope. That's code for client side (instruction to update battlefield and wait for user's click).